fcrepo-http-api/src/main/java/org/fcrepo/http/api/services/HttpRdfService.java

```diff
@@ -0,0 +1,157 @@
+import static org.fcrepo.kernel.api.rdf.DefaultRdfStream.fromModel;
+import static org.apache.jena.riot.RDFLanguages.contentTypeToLang;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.MediaType;
+import org.apache.jena.atlas.RuntimeIOException;
+import org.apache.jena.rdf.model.impl.StatementImpl;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RiotException;
+import org.fcrepo.http.commons.api.rdf.HttpIdentifierConverter;
+import org.fcrepo.kernel.api.exception.MalformedRdfException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.RdfStream;
+import org.slf4j.Logger;
+
+/**
+ * A service that will translate the resourceURI to Fedora ID in the Rdf InputStream
+ *
+ * @author bseeger
+ * @author bbpennel
+ * @since 2019-11-07
+ */
+
+public class HttpRdfService {
+
+    private static final Logger log = getLogger(HttpRdfService.class);
+
+    @Inject
+    private HttpIdentifierConverter idTranslator;
+
+    /**
+     * Parse the request body to a Model, with the URI to Fedora ID translations done.
+     *
+     * @param resourceId the ID of the Fedora resource
+     * @param stream the input stream containing the RDF
+     * @param contentType the media type of the RDF
+     * @return RdfStream containing triples from request body, with Fedora IDs in them
+     * @throws MalformedRdfException in case rdf json cannot be parsed
+     * @throws BadRequestException in the case where the RDF syntax is bad
+     */
+    public Model bodyToInternalModel(final String resourceId, final InputStream stream,
+                                     final MediaType contentType)
+                                     throws RepositoryRuntimeException, BadRequestException {
+
+        final Model model = parseBodyAsModel(stream, contentType, resourceId);
+        final List<Statement> insertStatements = new ArrayList<Statement>();
+        final StmtIterator stmtIterator = model.listStatements();
+
+        while (stmtIterator.hasNext()) {
+            final Statement stmt = stmtIterator.nextStatement();
+            final String originalSubj = stmt.getSubject().getURI();
+            final String subj = idTranslator.inExternalDomain(originalSubj) ?
+                idTranslator.toInternalId(originalSubj) : originalSubj;
+
+            RDFNode obj = stmt.getObject();
+            if (stmt.getObject().isResource()) {
+                final String objString = stmt.getObject().asResource().getURI();
+                obj = model.getResource(idTranslator.inExternalDomain(objString) ?
```

Comment:
if written without the ternary conditional you could probably save the system from having to do any work in the case of out of domain URIs:
```
if (idTranslator.inExternalDomain(objString)) {
  obj = model.getResource(objString);
}

Reply:
I have to translate it as well, but I think I get what you're saying here and will change it.

Reply:
Also it doesn't appear like we are translating predicates, that should probably be done as well?

Reply:
We haven't traditionally done translation of predicates, I think it part because they aren't references to Fedora resources. I'm okay with saving the system that extra work until we have a use case for it.